spec/plugins/redcarpet_spec.rb

```diff
@@ -13,7 +13,7 @@
   }
 
   let(:markdown) {
-    Redcarpet::Markdown.new(redcarpet.new, :fenced_code_blocks => true)
+    Redcarpet::Markdown.new(redcarpet.new({}), :fenced_code_blocks => true)
```

Comment:
??? is this not defaulted?

Reply:
My best guess is that the issue here is that the class that the `redcarpet` object represents is defined in C directly. If you look at [the documentation](https://www.rubydoc.info/gems/redcarpet/3.4.0/Redcarpet/Render/HTML) the implementing file is a C file and the constructor's source is a C function. I think the reference in the C function to `@options` is what the warning is picking up and complaint about when defined implicitly.

Reply:
I'm not... actually concerned at all about implicit instance variable usage, it's a feature of ruby that I rely on quite a bit and I'm a bit miffed that ruby warns about it to be honest.

Reply:
I don't think implicit instance variable usage is the problem per se. It's the way that it works with the C implementation that's causing the warning.

Reply:
Although, again, that's a best guess. This is the warning:

```
spec/plugins/redcarpet_spec.rb:16: warning: instance variable @options not initialized
```

Given this is line 16, that's rather oblique:

https://github.com/rouge-ruby/rouge/blob/904afdce5c43dc388130142dde4930c22df3cad0/spec/plugins/redcarpet_spec.rb#L15-L17

Reply:
@pyrmont I think this warning should be fixed at the upstream source itself instead in the test-suite here. Besides, this warning won't affect end-users of Jekyll, Gitlab since its not part of Rouge core.

Reply:
@ashmaroli The last release of Redcarpet was in December 2016. I wouldn't hold my breath on an update. We still have a circular require warning from their library that was [fixed in a patch](https://github.com/vmg/redcarpet/commit/25feb4eaf592f9a1ec5dceadf15416f66d292088) from January 2017 and that hasn't been released.

Reply:
I think this was fixed in 3.5.0, so the change may no longer be needed: https://github.com/vmg/redcarpet/commit/6b866560b25493049a52040932530de3a00cbac3